src/App.tsx

```diff
@@ -16,34 +17,132 @@
 ];
 
 export const App: React.FC = () => {
+  const [sortedGoods, setSortedGoods] = useState([...goodsFromServer]);
+  const [reversePressed, setReversePressed] = useState(false);
+  const [alphabeticallyPressed, setAlphabetically] = useState(false);
+  const [byLengthPressed, setByLength] = useState(false);
+
+  enum SortType {
+    Alphabetically = 'alphabetically',
+    ByLength = 'byLength',
+    Reverse = 'reverse',
+    Reset = 'reset',
+  }
+
+  const sortGoods = (sortType: SortType) => {
+    let sorted = [...sortedGoods];
+
+    switch (sortType) {
+      case 'alphabetically':
+        setAlphabetically(!alphabeticallyPressed);
+        if (byLengthPressed) {
+          setByLength(!byLengthPressed);
+        }
+
+        sorted.sort((a, b) => {
+          if (reversePressed) {
+            return a.localeCompare(b) * -1;
+          } else {
+            return a.localeCompare(b);
+          }
+        });
+        break;
+
+      case 'byLength':
+        setByLength(!byLengthPressed);
+        if (alphabeticallyPressed) {
+          setAlphabetically(!alphabeticallyPressed);
+        }
+
+        sorted.sort((a, b) => {
+          if (reversePressed) {
+            return (a.length - b.length) * -1;
+          } else {
+            return a.length - b.length;
+          }
+        });
+        break;
+
+      case 'reverse':
+        sorted.reverse();
+        setReversePressed(!reversePressed);
+        break;
+
+      case 'reset':
```

Comment:
The `case` statement should use `SortType.Reset` instead of the string 'reset'. This ensures consistency with the `enum` usage.